cmd/ui/src/views/Explore/ContextMenu/ContextMenu.tsx

```diff
@@ -14,82 +14,53 @@
-    const { primarySearch, secondarySearch, setExploreParams } = useExploreParams();
-    const { selectedItemQuery } = useExploreSelectedItem();
-    const { data: tierFlag } = useFeatureFlag('tier_management_engine');
+    onClose: () => void;
+    pathfindingFilters: PathfindingFilters;
+    position: MousePosition | null;
+}> = ({ onClose, pathfindingFilters, position }) => {
+    const { asEdgeItem, asNodeItem, exploreParams, isAssetGroupEnabled, menuPosition, selectedItemQuery } =
+        useContextMenuItems(position);
 
-    const ownedAssetGroupId = useAppSelector(selectOwnedAssetGroupId);
-    const tierZeroAssetGroupId = useAppSelector(selectTierZeroAssetGroupId);
+    const edgeItem = asEdgeItem(selectedItemQuery);
+    const nodeItem = asNodeItem(selectedItemQuery);
 
-    const { checkPermission } = usePermissions();
+    const tierZeroId = useAppSelector(selectTierZeroAssetGroupId);
+    const ownedId = useAppSelector(selectOwnedAssetGroupId);
 
-    const handleSetStartingNode = () => {
-        const selectedItemData = selectedItemQuery.data;
-        if (selectedItemData && isNode(selectedItemData)) {
-            const searchType = secondarySearch ? 'pathfinding' : 'node';
-            setExploreParams({
-                exploreSearchTab: 'pathfinding',
-                searchType,
-                primarySearch: selectedItemData?.objectId as string,
-            });
-        }
-    };
-
-    const handleSetEndingNode = () => {
-        const searchType = primarySearch ? 'pathfinding' : 'node';
-        const selectedItemData = selectedItemQuery.data;
-        if (selectedItemData && isNode(selectedItemData)) {
-            setExploreParams({
-                exploreSearchTab: 'pathfinding',
-                searchType,
-                secondarySearch: selectedItemData?.objectId as string,
-            });
-        }
-    };
+    if (!menuPosition || !(edgeItem || nodeItem)) {
+        return null;
+    }
 
     return (
-        <Menu
-            open={contextMenu !== null}
-            anchorPosition={{ left: contextMenu?.mouseX || 0 + 10, top: contextMenu?.mouseY || 0 }}
-            anchorReference='anchorPosition'
-            onClick={handleClose}>
-            <MenuItem onClick={handleSetStartingNode}>Set as starting node</MenuItem>
-            <MenuItem onClick={handleSetEndingNode}>Set as ending node</MenuItem>
+        <Menu open anchorPosition={menuPosition} anchorReference='anchorPosition' onClick={onClose}>
+            {edgeItem && <EdgeMenuItems id={edgeItem.id} pathfindingFilters={pathfindingFilters} />}
+
+            {nodeItem && <NodeMenuItems exploreParams={exploreParams} objectId={nodeItem.objectId} />}
+
+            {nodeItem && isAssetGroupEnabled && (
+                <>
+                    <AssetGroupMenuItem assetGroupId={tierZeroId} assetGroupName='High Value' />
+                    <AssetGroupMenuItem assetGroupId={ownedId} assetGroupName='Owned' />
+                </>
```

Comment:
smol thing but I get this error when this code block is used:
```
The Menu component doesn't accept a Fragment as a child.
Consider providing an array instead.
```

Ive never seen that before so I check it out and it doesnt appear to be breaking anything, just a check from MUI that a child isnt a fragment... perhaps using an array with these components as entries would quite the "error" down.